docs/source/howto/workflows.rst

```diff
@@ -20,6 +20,159 @@ Running a predefined workflow
+    If you want to write your own workflow, please see the section on :ref:`how to write your own workflows<how-to:workflows:write>`.
+
+Work functions
+--------------
+
+Running a work function is as simple as calling a typical Python function, simply call the function with the required input arguments:
+
+.. code-block:: python
+
+    from aiida.workflows.plugins.arithmetic.addmultiply import add_and_multiply
+    Int = DataFactory('int')
+
+    result = add_and_multiply(Int(2), Int(3), Int(5))
+
+Here, the ``add_and_multiply`` work function returns the output ``Int`` node and stores it in the variable ``result``.
+Note that the input arguments of a work function must be a subclass of the ``Data`` class.
+Just calling the ``add_and_multiply`` function with regular integers will result in a ``ValueError``, as these cannot be stored in the provenance.
+
+.. note::
+
+    Although the example above shows the most straightforward way to run the ``add_and_multiply`` work function, there are several other ways of running processes that can return more than just the result.
+    For example, the ``run_get_node`` function from the AiiDA engine returns both the result of the workflow and the work function node.
+    See the :ref:`corresponding topics section for more details <topics:processes:usage:launching>`.
+
+Work chains
+-----------
+
+To launch a work chain, you can either use the ``run`` or ``submit`` functions from the AiiDA engine.
+For either function, you need to provide the class of the work chain as the first argument, followed by the inputs as keyword arguments.
+Running a work chain means it is executed in the same interpreter in which it is launched:
+
+.. code-block:: python
+
+    from aiida.engine import run
+    Int = DataFactory('int')
+    MultiplyAndAddWorkChain = WorkflowFactory('arithmetic.multiplyadd')
+
+    add_code = load_code(label='add')
+
+    results = run(MultiplyAndAddWorkChain, x=Int(2), y=Int(3), z=Int(5), code=add_code)
+
+Alternatively, you can first construct a dictionary of the inputs, and pass it to the ``run`` function as an expanded dictionary:
+
+.. code-block:: python
+
+    inputs = {'x': Int(1), 'y': Int(2), 'z': Int(3), 'code': add_code}
+    results = run(MultiplyAndAddWorkChain, **inputs)
+
+This is particularly useful in case you have a workflow with a lot of inputs.
+In both cases, running the ``MultiplyAddWorkChain`` workflow returns the **results** of the workflow, i.e. a dictionary of the nodes that are produced as outputs, where the keys of the dictionary correspond to the labels of each respective output.
+
+.. note::
+
+    Similar to other processes, there are multiple functions for running a work chain. See the section on :ref:`launching processes for more details<topics:processes:usage:launching>`.
+
+Since *running* a workflow will block the interpreter, it is advisable to submit more complex work chains to the daemon:
```

Comment:
I would introduce this slightly different and first have the user realize the "problem" of running themselves. Only then we give the solution. So something like:

> You may have noticed that when you "run" a workflow, the interpreter is blocked, and you have to wait until the workflow is finished before you get back control. This might not be a problem for these trivial examples that only take a small amount of time to complete, however, when you start running longer workflows this will become problematic as you won't be able to turn your computer or even your terminal off until the workflow has fully terminated. Note that with this approach it is also difficult to run multiple workflows in parallel. The solution to these problems, is to, instead of running, submit the workflows to the daemon.

Then you show the code snippet and after you can continue with

> Note that when using `submit` the workflow is not run in the local interpreter but is sent off to the daemon and you get back control instantly. This allows you to submit multiple workflows at the same time and the daemon will start working on them in parallel. Once the `submit` call returns, you will not get the result as with `run`, but you will get the node that represents the workflow. This is because at that point in time, the results do not exist yet, because the workflow has not finished yet. In addition to allowing you to easily run multiple workflows in parallel, submitting workflows also allows you to shutdown your computer. The daemon will always save the progress of the workflows that is managing such that if it is stopped, it can restart where it left off


I see now that the text is quite long, but I think this is crucial for users to understand. See how you can fit it in and maybe shorten it a bit. Like I mentioned before that we will have the same problem in the calculation how-to, we might actually want to take part of this description and add it to a topic section and just reference that here, so we can reduce the amount of text here a bit. I would add it maybe to the same section that you [already referenced about the different launchers](https://aiida.readthedocs.io/projects/aiida-core/en/docs-revamp/topics/processes/usage.html#process-launch)

This reminds, we should also not forget that work functions *cannot* be submitted, but only work chains

Reply:
Be careful with the language here:

> You may have noticed that when you "run" a workflow, the interpreter is blocked

This sounds like a tutorial, however this is a how-to guide, not a tutorial. Users are not expected to actually execute this code to understand what is going on.

Reply:
I think I agree. Like I already alluded in the postscript, I think this should probably be explained once, well and in detail, probably in the "Topics - Processes" section. In the how-to, you can then just say that instead of running , we can also submit and then reference the section if you want to know when to use which and what are the pro's and con's of each. Would that work?

Reply:
I've shortened the pre-snippet paragraph and merged it with the text that was there. The post-snippet paragraph isn't that much longer than the text that was there, and expresses the important points more clearly, so I've largely replaced the text with your suggestion.

I think it's fine to leave it in the how-to, but in case you think it tries to explain too much, I can still shorten it.